test/scripts/validate-param-sys.py

```diff
@@ -0,0 +1,115 @@
+if test_setup.test_setup():
+
+    #############################################################################
+    # If this point was reached then begin testing parameter validation functions
+    #############################################################################
+
+    # Build a parameter table using bytearrays:
+    # Osc_A_Freq
+    Test_Packed_Table = bytearray(struct.pack(">f", 0.30))
+    # Osc_A_Amp
+    Test_Packed_Table += bytearray(struct.pack(">f", 5.00))
+    # Osc_A_Off
+    Test_Packed_Table += bytearray(struct.pack(">f", 2.50))
+    # Osc_B_Freq
+    Test_Packed_Table += bytearray(struct.pack(">f", 0.30))
+    # Osc_B_Amp
+    Test_Packed_Table += bytearray(struct.pack(">f", -5.00))
+    # Osc_B_Off
+    Test_Packed_Table += bytearray(struct.pack(">f", -2.50))
+    # Get table length:
+    Table_Length = len(Test_Packed_Table)
+    # Append table to Version and get CRC
+    CRC_Packed_Table = bytearray(struct.pack(">f", 0.0))
+    CRC_Packed_Table += Test_Packed_Table
+    Test_CRC = crc_16.crc_16(CRC_Packed_Table)
+    Int_CRC = int.from_bytes(Test_CRC, 'big')
+
+    # Send nominal Validate_Parameter_Table command expecting success:
+    cmd("Linux_Example", "Parameter_Manager_Instance-Validate_Parameter_Table", {
+          "Header.Table_Buffer_Length": Table_Length,
+          "Header.Crc_Table": Int_CRC,
+          "Table_Buffer": list(Test_Packed_Table)
+    })
+    # Check successful command count is 3 and that it was Validate_Parameter_Table with Status SUCCESS:
+    wait_check("Linux_Example Software_Status_Packet Command_Router_Instance.Command_Success_Count.Value == 3", 3)
+    wait_check("Linux_Example Software_Status_Packet Command_Router_Instance.Last_Successful_Command.Id == 5201", 3)
+    wait_check("Linux_Example Software_Status_Packet Parameter_Manager_Instance.Validation_Status.Last_Validation_Status == 'SUCCESS'", 3)
+    print("Validate_Parameter_Table SUCCESS OK")
+
+    # Send test Validate_Parameter_Table command with bad CRC expecting Memory_Region_Crc_Invalid, Table_Validation_Failure, and
+    # Command_Execution_Failure:
+    cmd("Linux_Example", "Parameter_Manager_Instance-Validate_Parameter_Table", {
+          "Header.Table_Buffer_Length": Table_Length,
+          "Header.Crc_Table": 0,
+          "Table_Buffer": list(Test_Packed_Table)
+    })
+    # Check last failed command count was 2 and that it was Validate_Parameter_Table with Status CRC_ERROR:
+    wait_check("Linux_Example Software_Status_Packet Command_Router_Instance.Command_Failure_Count.Value == 2", 3)
+    wait_check("Linux_Example Software_Status_Packet Command_Router_Instance.Last_Failed_Command.ID == 5201", 3)
+    print("Validate_Parameter_Table failure OK")
+    wait_check("Linux_Example Software_Status_Packet Command_Router_Instance.Last_Failed_Command.Status == 'FAILURE'", 3)
+    wait_check("Linux_Example Software_Status_Packet Parameter_Manager_Instance.Validation_Status.Last_Validation_Status == 'CRC_ERROR'", 3)
```

Comment:
Addressed in [bb3107a](https://github.com/lasp/adamant_example/pull/33/commits/bb3107a79ce639db92401058fc2b4540cc7ad37f). Doing this in a single call might require using `check_expression`, but it might be easier to check the software status packet buffer directly.